Mathlib/NumberTheory/LSeries/Dirichlet.lean

```diff
@@ -0,0 +1,420 @@
+
+
+section zeta
+
+/-!
+### The L-series of the constant sequence 1 / the arithmetic function ζ
+
+Both give the same L-series (since the difference in values at zero has no effect);
+see `ArithmeticFunction.LSeries_zeta_eq`, which agrees with the Riemann zeta function
+on `re s > 1`. We state most results in two versions, one for `1` and one for `↗ζ`.
+-/
+
+open LSeries Nat Complex DirichletCharacter
+
+/-- The abscissa of (absolute) convergence of the constant sequence `1` is `1`. -/
+lemma LSeries.abscissaOfAbsConv_one : abscissaOfAbsConv 1 = 1 :=
+  modOne_eq_one (χ := χ₁) ▸ absicssaOfAbsConv_eq_one one_ne_zero χ₁
+
+/-- The `LSeries` of the constant sequence `1` converges at `s` if and only if `re s > 1`. -/
+theorem LSeriesSummable_one_iff {s : ℂ} : LSeriesSummable 1 s ↔ 1 < s.re :=
+  modOne_eq_one (χ := χ₁) ▸ LSeriesSummable_iff one_ne_zero χ₁
+
+
+namespace ArithmeticFunction
+
+/-- The `LSeries` of the arithmetic function `ζ` is the same as the `LSeries` associated
+to the constant sequence `1`. -/
+lemma LSeries_zeta_eq : L ↗ζ = L 1 := by
+  ext s
+  refine LSeries_congr s fun n hn ↦ ?_
+  simp only [zeta_apply, hn, ↓reduceIte, cast_one, Pi.one_apply]
+
+/-- The `LSeries` associated to the arithmetic function `ζ` converges at `s` if and only if
+`re s > 1`. -/
+theorem zeta_LSeriesSummable_iff {s : ℂ} : LSeriesSummable (ζ ·) s ↔ 1 < s.re := by
+  have (n : ℕ) (hn : n ≠ 0) : ζ n = (1 : ℕ → ℂ) n := by
+    simp only [ArithmeticFunction.zeta_apply, hn, ↓reduceIte, Nat.cast_one, Pi.one_apply]
+  exact (LSeriesSummable_congr s this).trans <| LSeriesSummable_one_iff
+#align nat.arithmetic_function.zeta_l_series_summable_iff_one_lt_re ArithmeticFunction.zeta_LSeriesSummable_iff
+
+/-- The abscissa of (absolute) convergence of the arithmetic function `ζ` is `1`. -/
+lemma abscissaOfAbsConv_zeta : abscissaOfAbsConv ↗ζ = 1 := by
+  rw [abscissaOfAbsConv_congr (g := 1) fun n hn ↦ by simp [hn], abscissaOfAbsConv_one]
+
+/-- The L-series of the arithmetic function `ζ` equals the Riemann Zeta Function on its
+domain of convergence `1 < re s`. -/
+lemma LSeries_zeta_eq_riemannZeta {s : ℂ} (hs : 1 < s.re) : L ↗ζ s = riemannZeta s := by
+  simp only [LSeries, natCoe_apply, zeta_apply, cast_ite, CharP.cast_eq_zero, cast_one,
+    zeta_eq_tsum_one_div_nat_cpow hs]
+  refine tsum_congr fun n ↦ ?_
+  rcases eq_or_ne n 0 with rfl | hn
+  · simp only [term_zero, CharP.cast_eq_zero, ne_eq, ne_zero_of_one_lt_re hs, not_false_eq_true,
+      zero_cpow, div_zero]
+  · simp only [ne_eq, hn, not_false_eq_true, term_of_ne_zero, ↓reduceIte, one_div]
+
+/-- The L-series of the arithmetic function `ζ` equals the Riemann Zeta Function on its
+domain of convergence `1 < re s`. -/
+lemma LSeriesHasSum_zeta {s : ℂ} (hs : 1 < s.re) : LSeriesHasSum ↗ζ s (riemannZeta s) :=
+  LSeries_zeta_eq_riemannZeta hs ▸ (zeta_LSeriesSummable_iff.mpr hs).LSeriesHasSum
+
+lemma convolution_zeta_moebius : ↗ζ ⍟ ↗μ = δ := by
+  have hζ : ↗ζ = ↗(ζ : ArithmeticFunction ℂ) := rfl
```

Comment:
it looks weird to me to have this, but I guess there isn't a simp lemma one can write to deal with this in general.

Reply:
I have inlined this now in the next proof (which is the only place where it is used).